app/game/state_machine.js

```diff
@@ -769,10 +771,11 @@ StateMachine.prototype.buy_dev_card = function (data){
         data_package.player = this.game.players[data.player_id];
 
         //  Refreshes all player's scores, strip out to calc only one players score :- TODO
+        var longest_road_map = this.game.board.longest_roads(this.game.players);
         for (var p=0; p< this.game.players.length; p++) {
           var player_id = this.game.players[p].id;
-          this.game.players[p].score.longest_road = this.game.board.longest_road_for_player(player_id);
-        };
+          this.game.players[p].score.longest_road = longest_road_map.get(player_id);
+        }
```

Comment:
Is the section from line 774 -778 identical to 257 -261?
If so, can it be pulled out to it's own function?

Reply:
Yeah doing a fair clean up there.